### PR TITLE
Search Opt Out

### PR DESCRIPTION
**Pull request by clone1018**
This PR should add the is_searchable functionality, allowing users to control their privacy settings for our internal search and public search (Google, Yahoo, etc).

I also moved the metadata in the profile head that should be shared across all pages for that user.

### CODE REVIEW

File: tests/py/test_privacy_json.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        actual = json.loads(response.body)
+        assert actual['search_opt_out'] is False
+
+    def test_meta_robots_tag_added_on_opt_out(self):
+        apiResponse = self.hit_privacy('POST', data={'toggle': 'search_opt_out'})
+
+        expected = '<meta name="robots" content="noindex,nofollow" />'
+        response = self.client.GET("/alice")
+
+        assert expected in response.body
```
Comment (clone1018) [inline tests/py/test_privacy_json.py]: I need help getting this test to pass, will talk to @whit537

File: www/%username/privacy.json.spt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    out = {field: rec}
+else:
+    rec = website.db.one("""
+
+        SELECT search_opt_out AS search_opt_out
+        FROM participants WHERE username=%s
+
+    """, (user.participant.username,), back_as=dict)
+    assert rec is not None
+    out = rec
```
Comment (clone1018) [inline www/%username/privacy.json.spt]: Need help on this too from @whit537

File: branch.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+BEGIN;
+	ALTER TABLE participants ADD COLUMN is_searchable bool NOT NULL DEFAULT FALSE;
```
Comment (whit537) [inline branch.sql]: `DEFAULT TRUE`

File: templates/profile.html [hunk 4]
```diff
@@ -1,4 +1,28 @@
 {% extends "templates/participant.html" %}
+
+{% block head %}
+    <link rel="payment" type="text/html" title="Give weekly gift on Gratipay"
+        href="https://gratipay.com/{{ username }}/" />
+    <meta name="twitter:card" content="summary" />
+    <meta name="og:url" content="https://gratipay.com/{{ username }}/" />
```
Comment (whit537) [inline templates/profile.html]: `username` is `None` except for the actual profile page.

screen shot 2015-01-07 at 10 08 44 pm

File: templates/profile.html [hunk 5]
```diff
@@ -1,4 +1,28 @@
...
+    <meta name="og:type" content="profile" />
+    <meta name="og:title" content="{{ participant.get_og_title() }}" />
+    {% if statement %}
```
Comment (whit537) [inline templates/profile.html]: `statement` is also `None` except on profile.

File: templates/profile.html [hunk 6]
```diff
@@ -1,4 +1,28 @@
 {% extends "templates/participant.html" %}
+
+{% block head %}
+    <link rel="payment" type="text/html" title="Give weekly gift on Gratipay"
+        href="https://gratipay.com/{{ username }}/" />
```
Comment (whit537) [inline templates/profile.html]: `username` is `None` see below

**Pull request merged**